--- a/config.def.h
+++ b/config.def.h
@@ -3,12 +3,15 @@
 
 static int topbar = 1;                      /* -b  option; if 0, dmenu appears at bottom     */
 /* -fn option overrides fonts[0]; default X11 font or font set */
-static const int user_bh = 1;               /* add an defined amount of pixels to the bar height */
+static const int user_bh = 2;               /* add an defined amount of pixels to the bar height */
+static const int vertgap = 2;              /* space from top of screen */
+static const int horizgap = 2;             /* space from sides of screen */
+static const int horizpadbar = 2;          /* inner horizontal text pad (match dwm horizpadbar) */
 
 static const char *fonts[] = {
-  "JetBrains Mono:size=12:antialias=true:autohint=true"
+  "Noto Sans Mono:size=12:antialias=true:autohint=true"
 };
-static const char *prompt      = NULL;      /* -p  option; prompt to the left of input field */
+static const char *prompt      = "› ";      /* -p  option; prompt to the left of input field */
 
 static char normbgcolor[]           = "#010203";
 static char normfgcolor[]           = "#ccccff";
@@ -17,13 +20,19 @@ static char selbgcolor[]            = "#
 static char outfgcolor[]            = "#ccccff";
 static char outbgcolor[]            = "#010203";
 
+static const char cat_base[]        = "#1e1e2e"; // catppuccin mocha base
+static const char cat_surface1[]    = "#45475a"; // catppuccin mocha surface1
+static const char cat_mantle[]      = "#181825"; // catppuccin mocha mantle
+static const char cat_blue[]        = "#89b4fa"; // catppuccin mocha blue
+
+
 static const char *colors[SchemeLast][2] = {
-				/*     fg         bg       */
-	[SchemeNorm] = { normfgcolor, normbgcolor },
-	[SchemeSel]  = { selfgcolor,  selbgcolor  },
-	[SchemeSelHighlight] = { "#ffc978", "#005577" },
-	[SchemeNormHighlight] = { "#ffc978", "#222222" },
-	[SchemeOut]  = { outfgcolor,  outbgcolor },
+	/*                       fg             bg       */
+	[SchemeNorm] =          { cat_surface1, cat_mantle },
+	[SchemeSel]  =          { cat_blue,     cat_base  },
+	[SchemeSelHighlight] =  { "#000000",    cat_blue },
+	[SchemeNormHighlight] = { "#000000",    cat_surface1 },
+	[SchemeOut]  =          { outfgcolor,   outbgcolor },
 };
 /* -l option; if nonzero, dmenu uses vertical list with given number of lines */
 static unsigned int lines      = 0;
--- a/dmenu.c
+++ b/dmenu.c
@@ -705,9 +705,9 @@ setup(void)
 				if (INTERSECT(x, y, 1, 1, info[i]) != 0)
 					break;
 
-		x = info[i].x_org;
-		y = info[i].y_org + (topbar ? 0 : info[i].height - mh);
-		mw = info[i].width;
+		x = info[i].x_org + horizgap;
+		y = info[i].y_org + (topbar ? 0 : info[i].height - mh) + vertgap;
+		mw = info[i].width - 2*horizgap;
 		XFree(info);
 	} else
 #endif
@@ -715,9 +715,9 @@ setup(void)
 		if (!XGetWindowAttributes(dpy, parentwin, &wa))
 			die("could not get embedding window attributes: 0x%lx",
 			    parentwin);
-		x = 0;
-		y = topbar ? 0 : wa.height - mh;
-		mw = wa.width;
+		x = horizgap;
+		y = topbar ? vertgap : wa.height - mh + vertgap;
+		mw = wa.width - 2*horizgap;
 	}
 	promptw = (prompt && *prompt) ? TEXTW(prompt) - lrpad / 4 : 0;
 	inputw = mw / 3; /* input width: ~33% of monitor width */
@@ -826,7 +826,7 @@ main(int argc, char *argv[])
 	drw = drw_create(dpy, screen, root, wa.width, wa.height);
 	if (!drw_fontset_create(drw, fonts, LENGTH(fonts)))
 		die("no fonts could be loaded.");
-	lrpad = drw->fonts->h;
+	lrpad = drw->fonts->h + horizpadbar;
 
 #ifdef __OpenBSD__
 	if (pledge("stdio rpath", NULL) == -1)
